ui: Highlight unavailable ranges in red on the summary bar with nonzero

theloneexplorerquest opened the pull request:
Modify the summary bar to change the color of unavailable ranges. When the unavailable range is greater than zero, it will be displayed in red; if it is zero, it will be green.

Fix: #122014

Release note (ui): Changed the color of unavailable ranges on the summary bar to red when nonzero; ranges are green when zero.

cockroach-teamcity commented:
[![CLA assistant check](https://cla.crdb.dev/pull/badge/not_signed)](https://cla.crdb.dev/cockroachdb/cockroach?pullRequest=123120) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.crdb.dev/cockroachdb/cockroach?pullRequest=123120) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.crdb.dev/check/cockroachdb/cockroach?pullRequest=123120) it.</sub>

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/123120)
<!-- Reviewable:end -->

theloneexplorerquest commented on pkg/ui/workspaces/db-console/src/views/shared/components/summaryBar/summarybar.styl:
```diff
@@ -159,4 +162,4 @@
   &__label
     font-size 14px
     line-height 22px
-    letter-spacing 0.1px
```
my GoLand automatically remove the line here, I can revert if it is necessary.

abarganier replied:
Thanks for the explanation! If this isn't relevant/needed for the changes we're making in this PR, let's avoid changing it.

theloneexplorerquest commented on pkg/ui/workspaces/db-console/src/views/shared/components/summaryBar/index.tsx:
```diff
@@ -109,11 +109,17 @@ export function SummaryValue(
+  let classModifier = "number";
+
+  if (props.title === "Unavailable ranges" && props.value !== 0) {
+    classModifier = "alert-unavailable-ranges";
```
open to rename here

abarganier replied:
Name LGTM!

abarganier reviewed (changes requested):
Hello @theloneexplorerquest, welcome to CRDB!

Thanks for helping clean this up. I just have a couple small comments, but afterwards LGTM!

abarganier commented on pkg/ui/workspaces/db-console/src/views/shared/components/summaryBar/summarybar.styl:
```diff
@@ -100,6 +100,9 @@
   &--number &__value
     color $colors--primary-green-3
 
+  &--alert-unavailable-ranges  &__value
```
Sorry, this comment got lost in my original review. 

Can we change this class name so it's not specific to unavailable ranges? Maybe something like `number-alert`?

theloneexplorerquest replied:
> Sorry, this comment got lost in my original review.
> 
> Can we change this class name so it's not specific to unavailable ranges? Maybe something like `number-alert`?

Hi @abarganier Thanks for suggestion,
I have updated the code.

dhartunian commented on pkg/ui/workspaces/db-console/src/views/shared/components/summaryBar/index.tsx:
```diff
@@ -109,11 +109,17 @@ export function SummaryValue(
+  let classModifier = "number";
+
+  if (props.title === "Unavailable ranges" && props.value !== 0) {
+    classModifier = "alert-unavailable-ranges";
```
I'd prefer if we didn't have a special case hidden inside `SummaryStat`. This creates coupling between the mostly generic component and the specific use case. Imagine if we change the text of "unavailable ranges", then we'd have to remember to modify it here too. Ideally, the use case can communicate its intent in a more generic way.

Here are some options you can pick between:
* Create a `SummaryStatAlert` component with the special case.
* `SummaryStat` can take an additional prop that configures it to show the number in red. This could be a boolean named `alert` which matches Alex's CSS suggestion. When it's true, you can set the different class modifier.

theloneexplorerquest replied:
Hi @dhartunian Thanks for suggestion,
I have updated the code.

theloneexplorerquest commented on pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/summaryBar.tsx:
```diff
@@ -72,6 +72,7 @@ export const ClusterNodeTotals: React.FC = () => {
         </span>
       }
       value={nodeCounts.total}
+      numberAlert={false}
```
I checked in react boolean does not come with a default value, I think it is safer to set it as false?

dhartunian reviewed (commented):
@theloneexplorerquest changes look great! Thanks for iterating.

You will need to squash all your commits down into 1 before we can merge. Can you also make sure the PR description in github and the commit message match when you're done?

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/123120#-:-Nwpv_oZDEkpAJF5uXEw:bauq6ak)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @abarganier and @theloneexplorerquest)

___
*[`-- commits` line 4 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/123120#-NwpvOThBfkE_vZhwWeE):*
nit: can you wrap commit message at 70 characters?
___
*[`-- commits` line 8 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/123120#-NwpvRr42NZJRE7Nhw7B):*
nit: should say (ui change) in parentheses. Thanks for writing a release note!


<!-- Sent from Reviewable.io -->

theloneexplorerquest commented:
> @theloneexplorerquest changes look great! Thanks for iterating.
> 
> You will need to squash all your commits down into 1 before we can merge. Can you also make sure the PR description in github and the commit message match when you're done?
> 
> > _[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/123120#-:-Nwpv_oZDEkpAJF5uXEw:bauq6ak)_ status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @abarganier and @theloneexplorerquest)
> 
> _[`-- commits` line 4 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/123120#-NwpvOThBfkE_vZhwWeE):_ nit: can you wrap commit message at 70 characters?
> 
> _[`-- commits` line 8 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/123120#-NwpvRr42NZJRE7Nhw7B):_ nit: should say (ui change) in parentheses. Thanks for writing a release note!

Hi @dhartunian thanks for quick review, I have update the PR!

dhartunian reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/123120)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @abarganier and @theloneexplorerquest)

___
*[`pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/summaryBar.tsx` line 143 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/123120#-Nwym7nPDYihCOD8Yyyk:-Nwym7nQD-5g60G-YL6I:b-diwbhy) ([raw file](https://github.com/cockroachdb/cockroach/blob/3dc8eb62613a196306e3de86036a5e997d7c1090/pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/summaryBar.tsx#L143)):*
> ```XML
>           title="Unavailable ranges"
>           value={nodeSums.unavailableRanges}
>           numberAlert={nodeSums.unavailableRanges>0}
> ```

Tiny linter issue to put spaces around `>`

You can run `eslint` in the `pkg/ui/workspaces/db-console` directory via `pnpm run lint` or `./dev ui lint` in the root to run it via bazel to make sure it's clean before you push up.


<!-- Sent from Reviewable.io -->

theloneexplorerquest commented:
> > _[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/123120)_ status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @abarganier and @theloneexplorerquest)
> 
> _[`pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/summaryBar.tsx` line 143 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/123120#-Nwym7nPDYihCOD8Yyyk:-Nwym7nQD-5g60G-YL6I:b-diwbhy) ([raw file](https://github.com/cockroachdb/cockroach/blob/3dc8eb62613a196306e3de86036a5e997d7c1090/pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/summaryBar.tsx#L143)):_
> 
> > ```
> >           title="Unavailable ranges"
> >           value={nodeSums.unavailableRanges}
> >           numberAlert={nodeSums.unavailableRanges>0}
> > ```
> 
> Tiny linter issue to put spaces around `>`
> 
> You can run `eslint` in the `pkg/ui/workspaces/db-console` directory via `pnpm run lint` or `./dev ui lint` in the root to run it via bazel to make sure it's clean before you push up.

Thanks, updated 😄

dhartunian commented:
@theloneexplorerquest can you rebase on latest `master`, that should make the `unit_tests` job pass and then we can merge. thanks!

theloneexplorerquest commented:
> @theloneexplorerquest can you rebase on latest `master`, that should make the `unit_tests` job pass and then we can merge. thanks!

Thanks, git is tricky and took me a while to properly rebase. Hope it looks right, not sure why there are so many reviewers

dt commented:
I think you picked up a different commit authorship in the rebase; you might want to run `git commit --amend --reset-author --no-edit` and then force push the revised commit.

theloneexplorerquest committed:
ui: Highlight unavailable ranges in red on the summary bar

Color of unavailable ranges on the summary bar to be red when nonzero.

Fix: #122014

Release note (ui change): Color of unavailable ranges on the summary bar to be red when nonzero.

theloneexplorerquest commented:
> I think you picked up a different commit authorship in the rebase; you might want to run `git commit --amend --reset-author --no-edit` and then force push the revised commit.

Thanks, just updated

dhartunian commented:
bors r+

dhartunian commented:
@abarganier can you approve this PR and bors it? you have a change request lingering. thx!

abarganier reviewed (approved):
Sorry for the hold up! 

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/123120#-:-NyQsEUE5IkslepcRRf6:b-fajp15)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @dhartunian and @theloneexplorerquest)



<!-- Sent from Reviewable.io -->

abarganier commented:
bors r+
